ttnn/cpp/ttnn/operations/conv/conv2d/conv2d_utils.cpp

```diff
@@ -848,8 +955,366 @@ void adjust_conv_op_config_for_auto_shard_if_necessary(
+
+        uint32_t bias_block_num_bytes = per_core_out_matrix_width_ntiles * bias_tile_size;
+
+        uint32_t conv_act_c_blocks = weight_matrix_width_ntiles / per_core_out_matrix_width_ntiles;
+
+        uint32_t weight_block_w_ntiles = per_core_out_matrix_width_ntiles;
+        uint32_t weight_block_h_ntiles = act_block_w_ntiles;
+
+
+        uint32_t tilized_act_block_cb_size =  act_block_h_ntiles * act_block_w_ntiles * output_tile_size ;
+        uint32_t row_major_act_cb_size =  act_block_h_ntiles * act_block_w_ntiles * input_tile_size ;
+
+        uint32_t output_block_ntiles = per_core_out_matrix_height_ntiles * per_core_out_matrix_width_ntiles;
+
+        uint32_t num_blocks_act_w = weight_matrix_height_ntiles / act_block_w_ntiles;
+        uint32_t num_blocks_act_h = per_core_out_matrix_height_ntiles / act_block_h_ntiles;
+        uint32_t in0_num_blocks_w =
+            num_blocks_act_w * conv_act_c_blocks;  // Fold outer c_block loop together with weight_block_num_tiles = 9
+
+        packer_l1_acc = packer_l1_acc && ((enable_bias && in0_num_blocks_w > 1) || (in0_num_blocks_w > 2));
+
+        auto interm_dtype =
+            packer_l1_acc ? (fp32_dest_acc_en ? DataType::FLOAT32 : DataType::BFLOAT16) : output_dtype;
+
+        uint32_t partial_tile_size = tt::tile_size(datatype_to_dataformat_converter(interm_dtype));
+
+
+
+
+        //CB 0
+        uint32_t cb0_size = tilized_act_block_cb_size;
+        if(conv_config.enable_act_double_buffer) {
+            cb0_size *= 2;
+        }
+        tt::log_debug(tt::LogOp, "CB0 Size: {}", cb0_size);
+
+        //CB 1
+        uint32_t cb1_size = weight_block_h_ntiles * weight_block_w_ntiles * weights_tile_size;
+        if(conv_config.enable_weights_double_buffer) {cb1_size *= 2;}
+
+        tt::log_debug(tt::LogOp, "CB1 Size: {}", cb1_size);
+
+
+        //CB 2
+        uint32_t cb2_size = bias_block_num_bytes; tt::log_debug(tt::LogOp, "CB2 Size: {}", cb2_size);
+
+        //CB 5
+        uint32_t cb5_size = 64; tt::log_debug(tt::LogOp, "CB5 Size: {}", cb5_size);
+
+        //CB 6
+        uint32_t cb6_size = row_major_act_cb_size; tt::log_debug(tt::LogOp, "CB6 Size: {}", cb6_size);
+
+        //CB 24
+        uint32_t cb24_size = output_block_ntiles * partial_tile_size;
+        if(untilize_out==false && interm_dtype == output_dtype) {
+            cb24_size = 0;
+        } else {
+            tt::log_debug(tt::LogOp, "CB24 Size: {}", cb24_size);
+        }
+
+        //CB 25
+        uint32_t cb25_size = tilized_act_block_cb_size; tt::log_debug(tt::LogOp, "CB25 Size: {}", cb25_size);
+
+
+        uint32_t cb26_size = 0;
+        //CB26
+        if(untilize_out) {
+            cb26_size = weight_block_w_ntiles * output_tile_size; tt::log_debug(tt::LogOp, "CB26 Size: {}", cb26_size);
+        }
+
+        return{ output_size, cb0_size + cb1_size + cb2_size + cb5_size + cb6_size + cb24_size + cb25_size + cb26_size};
+    }
+    return {0, 0};
```

Comment:
should we throw here ?

Reply:
return {0, 0}; is now dead code it should be removed?

Reply:
Removed.